### PR TITLE
backup: expose backup service status

### PR DESCRIPTION
**Pull request by xiang90**
/cc @hongchaodeng @hasbro17 

I am going to add a method to get the status in the backup client library.

### CODE REVIEW

File: pkg/backup/backend.go [hunk 1]
```diff
@@ -17,7 +17,11 @@ package backup
 import "io"
 
 type backend interface {
-	save(version string, snapRev int64, rc io.Reader) error
+	save(version string, snapRev int64, rc io.Reader) (size int64, err error)
```
Comment (hongchaodeng) [inline pkg/backup/backend.go]: version -> etcdVersion
rc -> reader/r

mind add a comment to this method:
```
// save saves the snapshot from reader. The snapshot has given etcdVersion and snapRev.
// It returns the size of the snapshot saved.
save(...
```

File: pkg/backup/backupapi/types.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	// the backup service
+	RecentBackup *BackupStatus `json:"recentBackup,omitempty"`
+
+	// Backups is the totoal number of existing backups
+	Backups int `json:"backups"`
+}
+
+type BackupStatus struct {
+	// Size is the size of the backup.
+	Size int64 `json:"size"`
```
Comment (hongchaodeng) [inline pkg/backup/backupapi/types.go]: Also add revision?

File: pkg/backup/http.go [hunk 3]
```diff
@@ -34,6 +35,7 @@ const (
 func (b *Backup) startHTTP() {
 	http.HandleFunc(backupapi.APIV1+"/backup", b.serveSnap)
 	http.HandleFunc(backupapi.APIV1+"/backupnow", b.serveBackupNow)
+	http.HandleFunc(backupapi.APIV1+"/status", b.serveStatus)
```
Comment (hongchaodeng) [inline pkg/backup/http.go]: We also need to update backup_service.md . Are you going to do it in separate PR?

File: pkg/backup/backup.go [hunk 4]
```diff
@@ -198,7 +204,22 @@ func (b *Backup) writeSnap(m *etcdutil.Member, rev int64) error {
...
+		return err
+	}
+
+	bs := backupapi.BackupStatus{
+		Size:     n,
+		Version:  resp.Version,
+		TimeTook: time.Since(start),
+	}
+	b.recentBackupsStatus = append(b.recentBackupsStatus, bs)
+	if len(b.recentBackupsStatus) > 10 {
```
Comment (hongchaodeng) [inline pkg/backup/backup.go]: define const of 10.

File: pkg/backup/http.go (hunk 3 above)
Comment (xiang90) [inline pkg/backup/http.go]: i will update it in this pr.

File: pkg/backup/backupapi/types.go (hunk 2 above)
Comment (xiang90) [inline pkg/backup/backupapi/types.go]: not for now. i do not want to add more fields until we have a concrete use case.

**Comment by xiang90** [comment]
@hongchaodeng PTAL

**Comment by hongchaodeng** [comment]
lgtm
